misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/variable/types/BRecord.java

```diff
@@ -55,34 +64,67 @@ public Map<String, Value> computeChildVariables() {
             if (!(jvmValue instanceof ObjectReference)) {
                 return new LinkedHashMap<>();
             }
-            ObjectReference jvmValueRef = (ObjectReference) jvmValue;
-            Map<Field, Value> fieldValueMap = jvmValueRef.getValues(jvmValueRef.referenceType().allFields());
-            Map<String, Value> recordFields = new LinkedHashMap<>();
-            fieldValueMap.forEach((field, value) -> {
-                if (field.toString().contains(RECORD_FIELD_PATTERN_IDENTIFIER)) {
-                    recordFields.put(field.name(), value);
-                }
-            });
-            return recordFields;
+
+            Map<String, Value> childVarMap = new LinkedHashMap<>();
+            Map<Value, Value> recordFields = getRecordFields();
+
+            for (Map.Entry<Value, Value> mapEntry : recordFields.entrySet()) {
+                childVarMap.put(VariableUtils.getStringFrom(mapEntry.getKey()), mapEntry.getValue());
+            }
+            return childVarMap;
         } catch (Exception ignored) {
             return new LinkedHashMap<>();
         }
     }
 
+    private Map<Value, Value> getRecordFields() {
+        try {
+            Optional<Method> getKeysMethod = VariableUtils.getMethod(jvmValue,
+                                                                    METHOD_GET_KEYS,
+                                                                    GETKEYS_METHOD_SIGNATURE_PATTERN);
```

Comment:
Do we need to keep separate lines for each argument as we only have 3 args in here? :)